collect_app/src/main/java/org/odk/collect/android/activities/GoogleDriveActivity.java

```diff
@@ -795,18 +867,15 @@ void setGoogleDriveFormDownloadListener(GoogleDriveFormDownloadListener gl) {
 
                     String mediaDirName = FileUtils.constructMediaPath(fileItem.getName());
 
-                    String folderId;
+                    List<com.google.api.services.drive.model.File> mediaFileList;
                     try {
-                        folderId = driveHelper.getIDOfFolderWithName(mediaDirName, fileItem.getParentId(), false);
+                        mediaFileList = getMediaFiles(fileItem);
                     } catch (MultipleFoldersFoundException exception) {
                         results.put(fileItem.getName(), getString(R.string.multiple_media_folders_detected_notification));
                         return results;
                     }
 
-                    if (folderId != null) {
-                        List<com.google.api.services.drive.model.File> mediaFileList;
-                        mediaFileList = driveHelper.getFilesFromDrive(null, folderId);
-
+                    if (mediaFileList != null) {
                         FileUtils.createFolder(new StoragePathProvider().getDirPath(StorageSubdirectory.FORMS) + File.separator + mediaDirName);
```

Comment:
Please also check the entire file and use injected storagePathProvider instead of `new StoragePathProvider()` like here.

Reply:
Done.